OpenRA.Mods.Common/Widgets/Logic/GameSaveBrowserLogic.cs

```diff
@@ -201,7 +201,14 @@ void LoadGames(ScrollItemWidget gameTemplate, ScrollItemWidget newTemplate)
 				item.OnDoubleClick = Load;
 
 				var title = Path.GetFileNameWithoutExtension(savePath);
-				item.Get<LabelWidget>("TITLE").GetText = () => title;
+				var labelWidget = item.Get<LabelWithTooltipWidget>("TITLE");
+				var label = WidgetUtils.TruncateText(title, labelWidget.Bounds.Width, Game.Renderer.Fonts[labelWidget.Font]);
```

Comment:
This pattern could be pulled out into a `WidgetUtils.TruncatedLabelWithTooltip(LabelWidget label, string text)` to reduce duplication.

Reply:
I'm trying to do that but I'm getting an error:
```
The type or namespace name 'LabelWidget' could not be found (are you missing a using directive or an assembly reference?) (CS0246) [OpenRA.Game]
```

Reply:
I see. This is because `WidgetUtils` is in the `OpenRA.Game` project, while all the widget code is in `OpenRA.Mods.Common`.  It looks like `GetContrastColor` is the only helper used in `OpenRA.Game`, and that is only used from `SpriteFont`.  If you move that into a private helper inside `SpriteFont` then `WidgetUtils` can move to `OpenRA.Mods.Common` and this will then work.